seahub/api2/endpoints/repos_batch.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) 2012-2016 Seafile Ltd.
+import logging
+import json
+
+from django.http import HttpResponse
+from pysearpc import SearpcError
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+import seaserv
+from seaserv import seafile_api, ccnet_api
+
+from seahub.api2.authentication import TokenAuthentication
+from seahub.api2.throttling import UserRateThrottle
+from seahub.api2.utils import api_error
+from seahub.base.accounts import User
+from seahub.share.signals import share_repo_to_user_successful, \
+    share_repo_to_group_successful
+from seahub.utils import (is_org_context, send_perm_audit_msg)
+
+
+logger = logging.getLogger(__name__)
+json_content_type = 'application/json; charset=utf-8'
+
+class ReposBatchView(APIView):
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    permission_classes = (IsAuthenticated,)
+    throttle_classes = (UserRateThrottle,)
+
+    def get_repo_shared_to_users(self, request, repo_id):
+        username = request.user.username
+
+        if is_org_context(request):
+            org_id = request.user.org.org_id
+            share_items = seafile_api.list_org_repo_shared_to(org_id, username, repo_id)
+        else:
+            share_items = seafile_api.list_repo_shared_to(username, repo_id)
+
+        ret = []
+        for item in share_items:
+            ret.append(item.user)
+
+        return ret
+
+    def has_shared_to_user(self, request, repo_id, username):
+        users = self.get_repo_shared_to_users(request, repo_id)
+        has_shared = False
+        if username in users:
+            has_shared = True
+
+        return has_shared
+
+    def get_repo_shared_to_groups(self, request, repo_id):
+        username = request.user.username
+        if is_org_context(request):
+            org_id = request.user.org.org_id
+            share_items = seafile_api.list_org_repo_shared_group(org_id,
+                    username, repo_id)
+        else:
+            share_items = seafile_api.list_repo_shared_group_by_user(
+                    username, repo_id)
+
+        ret = []
+        for item in share_items:
+            ret.append(item.group_id)
+
+        return ret
+
+    def has_shared_to_group(self, request, repo_id, group_id):
+        group_ids = self.get_repo_shared_to_groups(request, repo_id)
+
+        has_shared = False
+        if group_id in group_ids:
+            has_shared = True
+
+        return has_shared
+
+    def post(self, request):
+
```

Comment:
op = request.GET.get('op')
        if not op or op != 'share':
            error_msg = 'op invalid.'
            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)